### PR TITLE
Upgrade to Python 3.7 & Sacred 0.7.5

### PR DESCRIPTION
**Pull request by AdamGleave**
Upgrade package versions. Fix some backward-incompatibilities with new version of Sacred.

### CODE REVIEW

File: src/modelfree/common/utils.py [hunk 1]
```diff
@@ -38,3 +38,17 @@ def add_artifacts(run, dirname, ingredient=None):
             relroot = os.path.relpath(path, dirname)
             name = prefix + relroot.replace('/', '_') + '_' + file
             run.add_artifact(path, name=name)
+
+
+def dict_deep_copy(d):
+    """Perform a deep copy on nested dictionaries.
+
+    Performs a shallow copy using `dict()`. Note this will *lose* subclasses,
```
Comment (decodyng) [inline src/modelfree/common/utils.py]: This documentation is a bit confusing between shallow and deep, and it doesn't look like we're actually using the dict conversion function anywhere in here

File: requirements-aprl.txt [hunk 2]
```diff
@@ -1,2 +1,3 @@
+git+git://github.com/HumanCompatibleAI/ilqr.git#egg=ilqr
```
Comment (decodyng) [inline requirements-aprl.txt]: What's going on here? This seems unrelated to the overall upgrade to 3.7

Comment (AdamGleave) [reply requirements-aprl.txt]: It was previously in `requirements.txt`, but I realised it's not needed in `modelfree`. Didn't seem worth making a separate PR for it...

File: src/modelfree/common/utils.py (hunk 1 above)
Comment (AdamGleave) [reply src/modelfree/common/utils.py]: Thanks, you're right this is confusing, addressed.

File: src/modelfree/common/utils.py [hunk 3]
```diff
@@ -43,12 +44,15 @@ def add_artifacts(run, dirname, ingredient=None):
 def dict_deep_copy(d):
     """Perform a deep copy on nested dictionaries.
 
-    Performs a shallow copy using `dict()`. Note this will *lose* subclasses,
-    e.g. OrderedDict. This is useful for e.g. `d` a Sacred read-only dict.
+    If `d` is an instance-of dict, copies `d` to a dict where the values are
+    recursively copied using `dict_deep_copy`. If `d` is not a dict, copies it
+    using `copy.deepcopy`. Note this intentioanlly loses subclasses.
```
Comment (decodyng) [inline src/modelfree/common/utils.py]: "intentionally" misspelled.

File: requirements-aprl.txt (hunk 2 above)
Comment (decodyng) [reply requirements-aprl.txt]: Makes sense

**Comment by AdamGleave** [comment]
There were a few additional bugs of the same kind. Will merge once tests pass.
